tests/derive_test.rs

```diff
@@ -102,6 +104,13 @@ enum NonStringKeys {
     Three { foo: u64 },
 }
 
+#[derive(Debug, Default, Clone, Eq, PartialEq, cbor::Encode, cbor::Decode)]
+struct Order {
+    second: u64,
+    first: u64,
+    thirdd: u64,
```

Comment:
any reason this field is not named `third`?

Reply:
Yeah, `third` would sort second while `thirdd` sorts third :-) This is due to how the canonical CBOR sort order is defined. It is defined in terms of the CBOR-encoded key and not the plain key and the CBOR-encoding of the key starts with the key's length, so a shorter key always sorts before a longer key.

Reply:
Makes sense, thanks!